scene/resources/resource_format_text.cpp

```diff
@@ -1398,6 +1399,65 @@ ResourceUID::ID ResourceLoaderText::get_uid(Ref<FileAccess> p_f) {
 	return ResourceUID::INVALID_ID;
 }
 
+// Returns empty string if the attached script could not be fetched.
+String ResourceLoaderText::get_attached_script_path(Ref<FileAccess> p_f) {
+	// Opening itself is pattern is also seen in resource_format_binary.cpp.
+	// It makes it convenient to not have to call open before this method.
+	open(p_f);
+
+	f = p_f;
+	stream.f = f;
+
+	ignore_resource_parsing = true;
+
+	while (next_tag.name == "ext_resource") {
+		if (!next_tag.fields.has("path")) {
+			return "";
+		}
+
+		if (!next_tag.fields.has("type")) {
+			return "";
+		}
+
+		if (!next_tag.fields.has("id")) {
+			return "";
+		}
+
+		String type = next_tag.fields["type"];
+
+		// Skip the tags until we find the "Script" tag
+		if (type == "Script") {
+			String path = next_tag.fields["path"];
+			String id = next_tag.fields["id"];
+
+			if (path.is_relative_path()) {
+				// Path is relative to file being loaded, so convert to a resource path.
+				path = ProjectSettings::get_singleton()->localize_path(local_path.get_base_dir().plus_file(path));
+			}
+
+			if (remaps.has(path)) {
+				path = remaps[path];
+			}
+
+			// Removed the parsed resources from the total
+			resources_total -= resource_current;
+			resource_current = 0;
+
+			return path;
+		}
+
+		error = VariantParser::parse_tag(&stream, lines, error_text, next_tag);
+
```

Comment:
@KoBeWi

> As I said, this is not important in most cases. When you create a custom resource the script will always be first

Did you perhaps mean that the script will always be *last*? Because I have a use case in my sample project for #48201 where an exported resource property itself uses a global script class that has another global script class property as its exported property, and in *that* use case, there are 2 `[ext_resource type="Script"...]` tags with the bottom one being the `script` property's `Script` (i.e. the "attached" script). Which then results in @Atlinx's code triggering a bug in my code.

To be specific, my demo has a `Tile2D` node that has a `terrain` property of type `Terrain`. It is possible to load a `WildTerrain` resource onto it (or an `UrbanTerrain`), and the `WildTerrain` has an exported array of `NaturalResource` resources. And in this use case, if I save e.g. a `wild_terrain_1.tres` file, that file needs both `NaturalResource` and `WildTerrain`, but it grabs the wrong one b/c `WildTerrain` is at the bottom and it stops after it finds the first one.

Either way, this PR needs to be amended to fix that.